### PR TITLE
docs: fix search bar focus on `Esc`

### LINKED ISSUE #16699: Docs: escape key would focus input
### Environment

N/A

### What parser are you using?

Other

### What did you do?

I just happened to press the `ESC` key by incident when browsing docs page https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new#configuring-shared-settings, and found it would scroll page to top and focus the search box. Which doesn't seem to be intentional.

Another problem is when you're focused in the search box, pressing `ESC` won't blur the focus from search box.

### What did you expect to happen?

1. When you browse web page, pressing `ESC` should do nothing when no input was focus
2. When you press `ESC` within search box, the search box should lose focus

### What actually happened?

See what did you do part.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Just curious, why not make use of [DocSearch](https://docsearch.algolia.com/docs/DocSearch-v3) library for the docs site? It already handled all those cases for us, and we can customize styles when needed.

### PR DESCRIPTION
**Pull request by shanpriyan**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Fixed search bar focusing on pressing `Esc`

#### Is there anything you'd like reviewers to focus on?

Resolves #16699

### CODE REVIEW

File: docs/src/assets/js/search.js [hunk 1]
```diff
@@ -171,10 +171,15 @@ if(searchClearBtn)
 
 document.addEventListener('keydown', function (e) {
 
+    const searchResults = Array.from(document.querySelectorAll('.search-results__item'));
+
     if (e.key === 'Escape') {
         e.preventDefault();
-        clearSearchResults();
-        searchInput.focus();
+
```
Comment (shanpriyan) [inline docs/src/assets/js/search.js:178]: remove new line
```suggestion
```

**Review by kecrily — APPROVED** [review]
LGTM, but I want someone to double check

**Review by harish-sethuraman — APPROVED** [review]
LGTM

**Comment by harish-sethuraman** [comment]
Thanks

**Review by amareshsm — COMMENTED** [review]
same here also https://github.com/eslint/eslint.org/pull/393#pullrequestreview-1229673525

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Comment by shanpriyan** [comment]
Shall I send a new PR with this change https://github.com/eslint/eslint.org/commit/1a281cea029d7abf83a35b2af749d464a3265bd5 for eslint.org/docs search bar?

Addresses 
> - When you press ESC within search box, the search box should lose focus

from https://github.com/eslint/eslint/issues/16699

**Pull request merged**